### PR TITLE
feat(config): Allow single strings for preset setting

### PR DESCRIPTION
**Pull request by zharinov**
Ref #7464

### CODE REVIEW

File: lib/config/migration.ts [hunk 1]
```diff
@@ -237,21 +237,28 @@ export function migrateConfig(
...
             isMigrated = true;
-            migratedConfig.extends[i] = 'config:js-app';
-          } else if (val[i] === ':library' || val[i] === 'config:library') {
+            preset = 'config:js-app';
+          } else if (preset === ':library' || preset === 'config:library') {
             isMigrated = true;
-            migratedConfig.extends[i] = 'config:js-lib';
-          } else if (is.string(val[i]) && val[i].startsWith(':masterIssue')) {
+            preset = 'config:js-lib';
+          } else if (is.string(preset) && preset.startsWith(':masterIssue')) {
```
Comment (viceice) [inline lib/config/migration.ts:257]: shouldn't the `is.string` check come first, that should be an error if it is not a string.

Comment (zharinov) [reply lib/config/migration.ts:257]: Seems like we don't validate preset values in general

File: lib/config/validation.ts [hunk 2]
```diff
@@ -214,17 +214,24 @@ export async function validateConfig(
...
+                    );
+                    if (!validTimezone) {
+                      errors.push({
+                        depName: 'Configuration Error',
+                        message: `${currentPath}: ${errorMessage}`,
+                      });
+                    }
                   }
+                } else {
+                  warnings.push({
```
Comment (zharinov) [inline lib/config/validation.ts:231]: Not sure if we should be strict on this or just ignore it with warning

Comment (viceice) [reply lib/config/validation.ts:231]: I think we should strict here, because we don't support other values.

File: lib/config/migration.ts (hunk 1 above)
Comment (rarkins) [reply lib/config/migration.ts]: I think they're validated later when we attempt to resolve them

**Review by rarkins — APPROVED** [review]

**Pull request merged**